lib/src/retries.dart

```diff
@@ -125,14 +129,15 @@ class _CircuitBreakerRetryPolicy implements RetryPolicy {
 
   _CircuitBreakerRetryPolicy({
     this.failureThreshold = 4,
+    this.maxAttempts = 20,
     this.resetTimeout = 10000,
     this.halfOpenThreshold = 2,
     this.coolDownTime = const (start: 1000, end: 2000),
   });
 
   @override
   FutureOr<bool> retry<T>(int attempts) async {
-    return await _transitionState(attempts);
+    return (await _transitionState(attempts) && attempts < maxAttempts);
```

Comment:
i think we should check for maxAttempts separately and then do a reset of the current state back to closed and the halfOpenAttempts back to 0.

if attempts > maxAttempts we should simply just return false, the condition for this shouldn't be hooked with the _transactionState.

Reply:
I just made a commit that address these concerns